### ISSUE TITLE
FEAT: User-specific connections

### ISSUE THREAD

**Issue by mstimvol** [dbgate/dbgate#605]
**Is your feature request related to a problem? Please describe.**
Currently, when a user adds a connection, all other users are seeing this connection as well.

**Describe the solution you'd like**
What we want is that each user has it's own connections. So when a user adds a connection, the connection isn't visible for other users.

**Describe alternatives you've considered**
Having separate installations of Dbgate, but that doesn't work for us because in this case we would need almost 100 separate installations.

**Comment by ProjectInfinity** [comment]
Could you describe your use-case?

How is DbGate installed and how is the computer accessed for you to have this problem?

**Comment by mstimvol** [comment]
@ProjectInfinity We're having a single database server and every user that is allowed to access the db has it's own databases. We want provide a graphical interface (Dbgate) for the users to connect to their own databases. The problem with Dbgate on our end is, when we have for example 10 users and one user logs in to Dbgate, adds his db connection, all other 5 users are seeing the database, too.

We're running Dbgate on our Linux server using Docker (see https://hub.docker.com/r/dbgate/dbgate).

**Comment by ProjectInfinity** [comment]
Yes of course, that is to be expected. There is no concept of "users" in DbGate, it's like running a desktop application in the cloud, the server becomes the "user" and everyone shares that "user".

In a situation like this I would suggest multiple docker containers and something custom to determine which docker container to connect to based on authentication. 

Sadly I think this kind of usage is a bit outside of the original scope of DbGate and it only applies to the web application since every platform's desktop application will be using the local user's configuration.

**Comment by mstimvol** [comment]
@ProjectInfinity Ok, Dbgate was a great alternative for us, but in this case yes... we need to setup multiple instances. As mentioned, it's not a very good solution since we need almost 100 different instances.

**Comment by tareksalem** [comment]
any plan to support this?

**Comment by ProjectInfinity** [comment]
There hasn't been any discussion in regard to supporting this and there are other issues with a higher priority at this time.

I totally understand the want and agree that for a shared Web UI it makes sense, but it is just not a priority at the moment.
The project is quite swamped with open issues, feel free to contribute if you are able to.
